apps/build.groovy

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env groovy
+
+/**
+ * <p>
+ * This script will be run by Jenkins when building this repository. Specifically, calling
+ * the <code>build()</code> method will build the Java applications: Data Pipeline, Data
+ * Server, etc.
+ * </p>
+ */
+
+
+/**
+ * Runs Maven with the specified arguments.
+ *
+ * @param args the arguments to pass to <code>mvn</code>
+ * @throws RuntimeException An exception will be bubbled up if the Maven build returns a non-zero exit code.
+ */
+def mvn(args) {
+	// This tool must be setup and named correctly in the Jenkins config.
+	def mvnHome = tool 'maven-3'
+
+	// Run the build, using Maven, with the appropriate config.
+	configFileProvider(
+			[
+				configFile(fileId: 'bfd:settings.xml', variable: 'MAVEN_SETTINGS'),
+				configFile(fileId: 'bfd:toolchains.xml', variable: 'MAVEN_TOOLCHAINS')
+			]
+	) {
+		sh "${mvnHome}/bin/mvn --settings $MAVEN_SETTINGS --toolchains $MAVEN_TOOLCHAINS ${args}"
+	}
+}
+
+/**
+ * Models the results of a call to {@link #build}: contains the paths to the artifacts that were built.
+ */
+class AppBuildResults implements Serializable {
```

Comment:
Just a note: I _think_ I can get away with embedding a class here, but I'm not entirely sure. Won't know either way until I actually try running this and see if/how it all blows up.

I might have to move these embedded classes (this and `AmiIds`) up to the top-level script, or worst-case replace them with a hokey `String[]` array.